File: KOD/judge.cpp
#include "judge.h"
#include <iostream>
#include <fstream>
#include <chrono>

using namespace std;

string testCases = "abcdef";
string testCasePrefix = "res/KOD0";
string testCaseSuffixIn = ".in";
string testCaseSuffixOut = ".out";

void judge::run(iKOD *k)
{
    for (auto cc : testCases) {
        auto begin = std::chrono::high_resolution_clock::now();

        ifstream ins(testCasePrefix +cc+ testCaseSuffixIn);
        ifstream outs(testCasePrefix +cc+ testCaseSuffixOut);

        int typ;
        ins >> typ;
        int n, codeLengths[20];
        ins >> n;
        for (int i = 0; i < n; ++i)
          ins >> codeLengths[i];

        long long expected;
        outs >> expected;
        auto res = k->countCodes(n, codeLengths);

        auto end = std::chrono::high_resolution_clock::now();

        if (res != expected) {
            cout << cc << " " << res << " " << expected << endl;
        } else {
            cout << cc << " " << res <<" OK " << chrono::duration_cast<chrono::milliseconds>(end - begin).count() << endl;
        }
    }
}

File: KOD/Judge.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;


public class Judge {

	static int mmax(int a, int b) { return a > b ? a : b; }

	static int task_rand(int n, int typ) {
		if (typ == 0)
			return pseudo_rand() % n;
		else if (typ == 1)
			return pseudo_rand() % mmax(1, n / 2);
		else if (typ == 2)
			return pseudo_rand() % mmax(1, n / 4);
		else if (typ == 3)
			return pseudo_rand() % mmax(1, n / 8);
		else
			return 0;
	}

	static public void run(KOD solution) {
		Scanner s = new Scanner(System.in);
		int typ = s.nextInt();
		int n;
		int[] codeLengths = new int[20];
		n = s.nextInt();
		for (int i = 0; i < n; ++i)
			codeLengths[i] = s.nextInt();
		long res = solution.countCodes(n, codeLengths);
		System.out.println(res);
	}

}

